### ISSUE TITLE
Rustfmt emits invalid code if a line is followed by a comment

### ISSUE THREAD

**Issue by iankronquist** [rust-lang-nursery/rustfmt#1292]
Consider this very basic Rust program:
```
fn main() {
    assert!(true); // Doesn't work without comment here.
    {
        println!("Do work in inner scope with guard object");
    }
}
```
When using rustfmt 0.7.0 with rustc 1.16.0-nightly (1a2ed98d3 2017-01-13) it emits this code:
```
ian@kartal:(~) → rustfmt test2.rs --write-mode diff
Diff in test2.rs at line 1:
 fn main() {⏎
-    assert!(true); // Doesn't work without comment here.⏎
+    assert!(true) // Doesn't work without comment here.⏎
     {⏎
         println!("Do work in inner scope with guard object");⏎
     }⏎
```

Notice how it removes a semi colon after the assertion.

When I try to compile the code rustfmt emits I get:
```
ian@kartal:(~) → rustfmt test2.rs
ian@kartal:(~) → rustc test2.rs
error: expected one of `.`, `;`, `?`, `}`, or an operator, found `{`
 --> test2.rs:3:5
  |
3 |     {
  |     ^

error: aborting due to previous error
```

If I move the comment to the previous line Rustfmt emits valid code.

**Comment by nrc** [comment]
This is fixed on master (just tested). It may also be fixed on 0.7.1, worth updating to that to check, though I don't have a version handy to test.

**Comment by iankronquist** [comment]
Appears to be fixed in 1.7.1
